main/guides/zoe/contract-basics.md

```diff
@@ -1,46 +1,56 @@
 
 <<< @/../snippets/zoe/src/01-hello.js#start
 
-Let's start with a contract with a simple `greet` function:
+Apart from `start`, we can have extra functions in our contract. These functions may be defined outside or within the `start` function depending on whether they need access to the variables defined in the `start` function.  For the greetings smart contract, we will have a simple `greet` function (outside `start`) that takes a string as a parameter (for example, name of the person calling the function) and returns a customized greeting message.
 
 <<< @/../snippets/zoe/src/01-hello.js#greet
 
-The `start` function can expose the `greet` function
-as part of the contract API by making it
-a method of the contract's `publicFacet`:
+The `greet` function, along with any other function besides `start`, must be exposed through the `start` function before it can be called. This is accomplished using a special object called `publicFacet`, which is returned by the `start` function. In the greeting contract, the `start` function exposes the `greet` function by defining it as a method of the contract's `publicFacet`, as shown below:
 
 <<< @/../snippets/zoe/src/01-hello.js#publicFacet
 
-We mark it `Far(...)` to allow callers to use it from outside the contract
-and give it a suggestive interface name for debugging.
+We wrap the reference to `greet` function in a `Far(...)` function call. This will allow callers to use it from outside the contract.
+We also give it a suggestive interface name `Hello` for debugging.
 _We'll discuss [Far in more detail](../js-programming/far) later._
 
 Putting it all together:
 
 <<< @/../snippets/zoe/src/01-hello.js#contract
 
+Let us save this code to a file named `01-hello.js` inside `src` directory. 
 ## Using, testing a contract
 
-Let's use some tests to explore how a contract is used.
+Before we deploy or start our contract, it is usually a good idea to run a few tests that validate its correctness. Agoric contracts are typically tested using the [ava](https://github.com/avajs/ava) framework. They start with `@endo/init` to establish a [Hardened JavaScript](../js-programming/hardened-js) environment as below:
 
-Agoric contracts are typically tested
-using the [ava](https://github.com/avajs/ava) framework.
-They start with `@endo/init` to establish a [Hardened JavaScript](../js-programming/hardened-js) environment:
+<<< @/../snippets/zoe/contracts/test-zoe-01-hello.js#test-imports
 
-<<< @/../snippets/zoe/contracts/test-zoe-hello.js#test-imports
+We also import `E()` in order to make asynchronous method calls and `test` function from `ava`. _We'll talk more about [using `E()` for async method calls](../js-programming/eventual-send) later._ Also note the `eslint-disable-next-line` command in the comment on line 3; this is needed in order to suppress an error that arises due to a known issue in `ava`.
```

Comment:
command -> comment

Reply:
Are you referring to the `eslint-disable-next-line`?  Let us talk about it. I think it should as is but I will fix it if you think otherwise.